docs/v1-proposed-api.md

```diff
@@ -0,0 +1,11 @@
+# Proposed Changes for Hive V1 API
+
+ 1. Require an InstallConfig ConfigMap, referenced in a new ClusterDeployment.Spec field, to pass directly to installer.
+   * Remaining ClusterDeployment.Spec should be mostly, things that can be modified post-install.
+   * Would allow quicker exposure to new Installer features without needing to mirror in Hive.
+   * Must watch out for secrets (pull secret?) that should not be stored in ConfigMap. If it's just PullSecrets, this might be ok, but otherwise we may need to have Hive logic to inject these.
```

Comment:
One option would be to make the installconfig a template that gets populated with values from secrets. The user could hide anything that they wanted to hide in secrets. Things like pull secret and ssh key would still be a bit trickier since they are used explicitly by hive instead of being passed blindly along to the installer.